### PR TITLE
First try

### PR DESCRIPTION
**Pull request by SerhiiUnhurian**
[DEMO LINK](https://SerhiiUnhurian.github.io/react_todo-app-loading-todos/)

### CODE REVIEW

**Review by sTorba24 — CHANGES_REQUESTED** [review]
Good work! Please provide DEMO link to check your work

File: src/components/ErrorMessages/ErrorMessages.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import React, { useContext } from 'react';
+import classNames from 'classnames';
+import { TodosContext } from '../TododsContext/TodosContext';
+
+export const ErrorMessages: React.FC = () => {
+  const { errorMessage, setErrorMessage } = useContext(TodosContext);
+
+  let isErrorHidden = true;
+
+  if (errorMessage !== '') {
```
Comment (sTorba24) [inline src/components/ErrorMessages/ErrorMessages.tsx:10]:
```suggestion
  if (errorMessage) {
```

File: src/components/ErrorMessages/ErrorMessages.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    isErrorHidden = false;
+    setTimeout(() => {
+      setErrorMessage('');
+      isErrorHidden = true;
+    }, 3000);
+  }
+
+  return (
+    // {/* Notification is shown in case of any error */}
+    // {/* Add the 'hidden' class to hide the message smoothly */}
```
Comment (sTorba24) [inline src/components/ErrorMessages/ErrorMessages.tsx:20]: remove comments

Comment (sTorba24) [reply src/components/ErrorMessages/ErrorMessages.tsx:20]: fix everywhere

File: src/components/Footer/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { Filter } from '../Filter/Filter';
+
+export const Footer: React.FC = () => {
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        3 items left
```
Comment (sTorba24) [inline src/components/Footer/Footer.tsx:8]: it should be dynamic

File: src/components/TodoList/TodoList.tsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      //   <button type="button" className="todo__remove" data-cy="TodoDelete">
+      //     ×
+      //   </button>
+
+      //   {/* 'is-active' class puts this modal on top of the todo */}
+      //   <div data-cy="TodoLoader" className="modal overlay is-active">
+      //     <div className="modal-background has-background-white-ter" />
+      //     <div className="loader" />
+      //   </div>
+      // </div>
```
Comment (sTorba24) [inline src/components/TodoList/TodoList.tsx:136]: remove comments

**Review by sTorba24 — CHANGES_REQUESTED** [review]

File: src/components/TodoApp.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  const { todos } = useContext(TodosContext);
+
+  return (
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header />
+
+        {todos.length > 0 && (
```
Comment (volodymyr-soltys97) [inline src/components/TodoApp.tsx:18]:
```suggestion
        {!!todos.length && (
```

File: src/components/TodoList/TodoList.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+            <input
+              data-cy="TodoStatus"
+              type="checkbox"
+              className="todo__status"
+              checked={todo.completed}
+            />
+          </label>
+
+          <span data-cy="TodoTitle" className="todo__title">
+            {todo.title}
```
Comment (volodymyr-soltys97) [inline src/components/TodoList/TodoList.tsx:48]: Use a destruction for `todo`

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Good job 👍 
Let's improve your code

**Review by volodymyr-soltys97 — APPROVED** [review]
Great work 🔥

**Pull request closed without merging**